docs/source/examples/flow_past_cylinder.rst

```diff
@@ -0,0 +1,252 @@
+created, and manager class are returned. The ``inlet_info`` and ``outlet_info``
+info contains specific information about inlet and outlet that enables IOM to
+create equations, stepper and updater. In ``_create_inlet_outlet_manager``
+the IOM is created using the info objects.
+
+Passing IOM to scheme
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: python
+
+    def configure_scheme(self):
+        scheme = self.scheme
+        self.iom = self._create_inlet_outlet_manager()
+        scheme.inlet_outlet_manager = self.iom
+        pfreq = 100
+        kernel = QuinticSpline(dim=2)
+        self.iom.update_dx(self.dx)
+        scheme.configure(h=self.h, nu=self.nu)
+
+        scheme.configure_solver(kernel=kernel, tf=self.tf, dt=self.dt,
+                                pfreq=pfreq, n_damp=0)
+
+
+The IOM object of the application is initialized in the method
+``configure_scheme`` of the ``Application`` class. All the
+post-initialization method could be called here e.g. ``update_dx``.
+
+Creating ghost particles
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this example, we have a wind tunnel with two bounding slip walls on the
+top and below of the tunnel. The inlet is on the left and the outlet is on
+the right. In order to perform the simulation five particle arrays,
+``solid``, ``fluid``, ``wall``, ``inlet`` and ``outlet`` are generated.
+``fluid``, ``solid`` and ``wall`` has to obey ``edac`` scheme, whereas
+``inlet`` and ``outlet`` are solved according to the boundary conditions.
+
+.. code:: python
+
+    def create_particles(self):
+        fluid = self._create_fluid()
+        solid = self._create_solid()
+        outlet = self._create_outlet()
+        inlet = self._create_inlet()
+        wall = self._create_wall()
+
+        ghost_inlet = self.iom.create_ghost(inlet, inlet=True)
+        ghost_outlet = self.iom.create_ghost(outlet, inlet=False)
+
+        particles = [fluid, inlet, outlet, solid, wall]
+        if ghost_inlet:
+            particles.append(ghost_inlet)
+        if ghost_outlet:
+            particles.append(ghost_outlet)
+
+        self.scheme.setup_properties(particles)
+        self._set_wall_normal(wall)
+
+        if self.io_method == 'hybrid':
+            fluid.uag[:] = umax
+            fluid.uta[:] = umax
+            outlet.uta[:] = umax
+
+        return particles
+
+The particle arrays ``ghost_inlet`` and ``ghost_outlet`` are generated by
+the IOM depending upon the type of IOM subclass used. The properties
+:math:`uag`, :math:`uta` are the time average and velocity array in :math:`x`
+direction at t=0.
+
+Creating updater
+~~~~~~~~~~~~~~~~~
+
+The purpose of the updater is to remove particle from ``inlet`` and add them
+to ``fluid`` whenever a particle crosses the inlet-outlet interface. Similarly,
+it is done in case of the ``oulet``. It also adds new particle to ``inlet`` as required and remove a particle from the simulation when they flow past ``outlet``.
```

Comment:
Is this done automatically or does the user need to write anything extra?

Reply:
yes done automatically. Added that in the doc.